FilePersistence/src/version_control/Commit.cpp

```diff
@@ -71,18 +71,37 @@ void Commit::addFile(QString relativePath, qint64 size, std::unique_ptr<char[],
 	files_.insert(relativePath, new CommitFile{relativePath, size, std::move(content)});
 }
 
-bool Commit::getFileContent(QString fileName, const char*& content, int& contentSize) const
+bool Commit::getFileContent(QString fileName, const char*& content, int& contentSize, bool exactFileNameMatching) const
 {
-	QHash<QString, CommitFile*>::const_iterator iter = files_.find(fileName);
-	if (iter != files_.constEnd())
+	// name of file must match fileName exactly
+	if (exactFileNameMatching)
 	{
-		contentSize = iter.value()->size_;
-		content = iter.value()->content();
+		QHash<QString, CommitFile*>::const_iterator iter = files_.find(fileName);
+		if (iter != files_.constEnd())
+		{
+			contentSize = iter.value()->size_;
+			content = iter.value()->content();
 
-		return true;
+			return true;
+		}
 	}
+	// name of file contains fileName
 	else
-		return false;
+	{
+		QHash<QString, CommitFile*>::const_iterator iter = files_.constBegin();
+		while (iter != files_.constEnd())
+		{
+			if (iter.key().contains(fileName))
+			{
```

Comment:
Always avoid code duplication. The body of this if is identical to the body above. Define `iter` at the very beginning of the method, then use one or the other way to set its value and at the end check if that's different than `end` and if so put the body of this if to set the return values.